Reference/Routing/custom-routes.md

```diff
@@ -300,6 +296,66 @@ namespace RoutingDocs.Controllers
 
 With that we have our controller with a custom route within an Umbraco context.
 
+#### Client Side Requests
+If the endpoint of your custom route is considered a client-side request e.g. **/sitemap.xml**, you will need make a few of changes to get this to work.
+
+Define your route as before, specifying the correct client type route:
+
+```
+.WithEndpoints(u =>
+{
+    u.EndpointRouteBuilder.MapControllerRoute("Sitemap Xml", "/sitemap.xml",
+        new { Controller = "SitemapXml", Action = "Index" });
+});
+```
+
+You will need to configure your route request options within your **Startup.cs** class. For single routes:
+
+```
+services.Configure<UmbracoRequestOptions>(options =>
+{
+    options.HandleAsServerSideRequest = httpRequest => httpRequest.Path.StartsWithSegments("/sitemap.xml");
+});
+```
+
+Or it can handle multiple routes:
+
+```
+services.Configure<UmbracoRequestOptions>(options =>
+{
+    string[] allowList = new[] {"/sitemap.xml", ...};
+    options.HandleAsServerSideRequest = httpRequest =>
+    {
+        foreach (string route in allowList)
+        {
+            if (httpRequest.Path.StartsWithSegments(route))
+            {
+                return true;
+            }
+        }
+
+        return false;
+    };
+});
+```
+In your **FindContent** method you should still be able to access and use **IUmbracoContextAccessor** through standard DI:
+
+```
+public IPublishedContent FindContent(ActionExecutingContext actionExecutingContext)
+{
+    IPublishedContent content = default(IPublishedContent);
+    var context = _umbracoContextAccessor.GetRequiredUmbracoContext();
+
+    content = context.Content.GetAtRoot().FirstOrDefault();
+
+    return content;
+}
```

Comment:
```suggestion
public IPublishedContent? FindContent(ActionExecutingContext actionExecutingContext)
{
    IUmbracoContext context = _umbracoContextAccessor.GetRequiredUmbracoContext();
    IPublishedContent? content = context.Content?.GetAtRoot().FirstOrDefault();

    return content;
}
```

I'd recommend changing this method up a bit to something like this. 

This way these docs will work for both v9 and v10, even though v9 complains with "The annotation for nullable reference types should only be used in code within a '#nullable' annotations context", however this works just fine still. 

Fun bonus fact, reviewing this showed us a nullability bug in v10 that I've fixed here: https://github.com/umbraco/Umbraco-CMS/pull/12545/ so double H5YR :D

Reply:
LGTM and happy with the changes if it makes sense for v10 docs